src/wp-includes/formatting.php

```diff
@@ -2138,7 +2138,17 @@ function sanitize_user( $username, $strict = false ) {
 
 	// If strict, reduce to ASCII for max portability.
 	if ( $strict ) {
-		$username = preg_replace( '|[^a-z0-9 _.\-@]|i', '', $username );
+		// Removed @ to disallow email adresses as username for new users.
+		$allowed_chars = 'a-z0-9 _.\-';
+
+		/**
+		 * Filters the regexp of characters which are allowed in a username in strict mode.
+		 *
+		 * @param string   $allowed_chars Regexp pattern for matching allowed characters.
+		 * @param string   $raw_username  The original username to be sanitized.
```

Comment:
```suggestion
		 *
		 * @since x.x.x
		 *
		 * @param string $allowed_chars Regexp pattern for matching allowed characters.
		 * @param string $raw_username  The original username to be sanitized.
```
- Let's add a placeholder `@since` so that this isn't missed.
- Removing some extra spaces after the parameter types.